custom_enumerable_methods.rb

```diff
@@ -0,0 +1,119 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    my_self = []
+    i = 0
+    while i < my_self.size
+      yield(my_self[i])
+      i += 1
+    end
+    my_self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_self = []
+    (i...my_self.size).my_each do |i|
+      yield(my_self[i], i)
+    end
+    my_self
+  end
+
+  def my_select
+    result = []
+    my_self = []
+    my_self.my_each { |i| result.push(i) if yield(i) }
+    result
+  end
+
+  def my_all?
+    result = true
+    my_self = []
+    my_self.my_each { |i| false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_self = []
+    my_self.my_each { |i| false unless yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_self = []
+    my_self.my_each do |i|
+      result = false if my_self[i] == yield(i)
+    end
+    result
+  end
```

Comment:
```rb
2.6.5 :096 > false_array = [nil, false, true, []]
2.6.5 :097 > false_array.none?
 => false 
2.6.5 :098 > false_array.my_none?
 => true 
2.6.5 :099 >
```

Reply:
```rb
2.6.5 :105 > true_block = proc { |num| num <= 9 }
2.6.5 :106 > array.none?(&true_block)
 => false 
2.6.5 :107 > array.my_none?(&true_block)
 => true 
2.6.5 :108 >
```

Reply:
```rb
2.6.5 :110 > array
 => [5, 4, 4, 0, 5, 0, 3, 4, 0, 0, 2, 6, 5, 6, 5, 6, 2, 1, 3, 0, 8, 0, 0, 1, 7, 1, 3, 5, 6, 1, 1, 1, 6, 5, 7, 7, 2, 0, 3, 1, 7, 0, 1, 4, 5, 7, 6, 5, 4, 3, 6, 5, 4, 0, 6, 2, 7, 2, 5, 6, 4, 7, 1, 3, 6, 3, 3, 2, 6, 1, 7, 7, 8, 6, 1, 3, 5, 0, 4, 1, 0, 5, 4, 2, 5, 2, 5, 6, 6, 0, 1, 8, 4, 5, 4, 0, 4, 0, 8, 6] 
2.6.5 :111 > array.my_none?(String)
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        3: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        2: from (irb):111
        1: from custom_enumerable_methods.rb:45:in `my_none?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :112 > array.none?(String)
 => true 
2.6.5 :113 > 
```

Reply:
# Thank You! :bouquet: 
## Above issue is fixed now! :heavy_check_mark: 

## I helped us to learn more and fix the things.